### PR TITLE
Added Basecamp API example

### PR DESCRIPTION
**Pull request by kospl**

### CODE REVIEW

File: samples/Basecamp.gs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  service.reset();
+}
+
+/**
+ * Configures the service.
+ */
+function getService() {
+  return OAuth2.createService('Basecamp')
+      // Set the endpoint URLs.
+      .setAuthorizationBaseUrl('https://launchpad.37signals.com/authorization/new?type=web_server')
```
Comment (oshliaer) [inline samples/Basecamp.gs]: I'm sorry but it's not transparent. Could you add something like a comment with

1. Register your app at integrate.37signals.com. You'll be assigned a client_id and client_secret. You'll need to provide a redirect_uri: a URL where we can send a verification code. Just enter a dummy URL like http://myapp.com/oauth if you're not ready for this yet.
1. Configure your OAuth 2 library with your client_id, client_secret, and redirect_uri. Tell it to use https://launchpad.37signals.com/authorization/new to request authorization and https://launchpad.37signals.com/authorization/token to get access tokens.

Comment (erickoledadevrel) [inline samples/Basecamp.gs]: I would avoid adding this in. These samples are meant to show how to configure the library with the given OAuth provider, but they shouldn't get into the details of how to register an application, get keys, etc. Those details are best left to the provider's own documentation, where it is more likely to stay up-to-date. Perhaps we just link off to their documentation?

Comment (oshliaer) [inline samples/Basecamp.gs]: It's seems better. I found the docs here https://github.com/basecamp/api/blob/master/sections/authentication.md

**Comment by erickoledadevrel** [comment]
Thanks for adding the sample!

**Pull request merged**